relay/ethindex_db.py

```diff
@@ -0,0 +1,311 @@
+    Since the proxy classes operates on one network address only,
+    we allow to pass a default address in.
+    """
+
+    def __init__(self, conn, standard_event_types, event_builders, from_to_types, address=None):
+        self.conn = conn
+        self.default_address = address
+        self.standard_event_types = standard_event_types
+        self.event_builder = EventBuilder(event_builders)
+        self.from_to_types = from_to_types
+
+    @property
+    def event_types(self):
+        return self.event_builder.event_types
+
+    def _build_events(self, rows):
+        return self.event_builder.build_events(rows, self._get_current_blocknumber())
+
+    def _get_current_blocknumber(self):
+        with self.conn.cursor() as cur:
+            cur.execute("""select * from sync where syncid='default'""")
+            row = cur.fetchone()
+            if row:
+                return row["last_block_number"]
+            else:
+                raise RuntimeError("Could not determine current block number")
+
+    def _get_addr(self, address):
+        """all the methods here take an address argument
+        We use the default address instead. Eventually callers will
+        need to provide this argument, and we can remove the
+        default_network_address.
+        """
+        r = address or self.default_address
+        assert r, "no network address passed in and no default network address given"
+        return r
+
+    def _get_standard_event_types(self, standard_event_types):
+        r = standard_event_types or self.standard_event_types
+        assert r, "no standard event passed in and no default events given"
+        return r
+
+    def _run_events_query(self, events_query: EventsQuery) -> List[BlockchainEvent]:
+        """run a query on the events table"""
+        query_string = "{select_star_from_events} WHERE {where_block} {order_by_default_sort_order}".format(
+            select_star_from_events=select_star_from_events,
+            where_block=events_query.where_block,
+            order_by_default_sort_order=order_by_default_sort_order)
+
+        with self.conn as conn:
+            with conn.cursor() as cur:
+                cur.execute(query_string, events_query.params)
+                rows = cur.fetchall()
+                return self._build_events(rows)
+
+    def get_network_events(
+            self,
+            event_name: str,
+            user_address: str = None,
+            from_block: int = 0,
+            timeout: float = None,
+            network_address: str = None,
+    ) -> List[BlockchainEvent]:
+        return self.get_user_events(
+            event_name,
+            user_address,
+            from_block,
+            timeout,
+            network_address,
+        )
+
+    def get_unw_eth_events(
+            self,
+            event_name: str,
+            user_address: str = None,
+            from_block: int = 0,
+            timeout: float = None,
+            network_address: str = None,
```

Comment:
Variable name `network_address` is a bit confusing. Change to `unw_eth_address`?

Reply:
Yes. That was a copy past error, Agreed.

Reply:
Actually we can just remove it, as these functions are there only for compatibility.